samples/matter/common/src/event_manager.cpp

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include "event_manager.h"
+#include "app_event.h"
+#include "system_event.h"
+
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+
+#include <platform/Zephyr/SysHeapMalloc.h>
+
+LOG_MODULE_DECLARE(app, CONFIG_CHIP_APP_LOG_LEVEL);
+
+constexpr size_t kEventQueueSize = 10;
+
+K_MSGQ_DEFINE(sEventQueue, sizeof(Event), kEventQueueSize, alignof(Event));
+
+void EventManager::PostEvent(Event &event)
+{
+	/* Allocate context on the Heap */
+	void* context = chip::DeviceLayer::Malloc::Malloc(event.mContextSize);
+	if(!context){
+		LOG_INF("Failed to store context in Heap");
```

Comment:
All methods that use the PostEvent method in the BoardInterface are void, but yeah it will be ok to implement it this way from the user's perspective.